incorporate Hang's suggestions, and also remove all Metadata tables, focus use cases on signaling (#16)
danwing/signaling-use-cases: draft-rwbr-tsvwg-signaling-use-cases.md

The client-centric metadata sharing approach because it preserves privacy and also
takes advantage of clients having a full view on their available network attachments.
Without client involvement some use-cases cannot be solved, as detailed below

> in {{generic-cases}}.

Reviewer: ```suggestion
in {{generic-cases}}. 

> Note: some use cases may require seeking for users preference or receiving these preferences. Such matters are
out of scope.
```

is value in differentiating between voice, video and data.  Today's
video streaming is exclusively over TCP but will migrate to QUIC and
eventually is likely to support unreliable transport ({{?RFC9221}},

> {{?I-D.draft-ietf-moq-transport}}).  With unreliable transport of video in

Reviewer: ```suggestion
{{?I-D.ietf-moq-transport}}).  With unreliable transport of video in
```

Streaming video also contains audio frames which can be encoded
separately and thus can be signaled separately.  Audio is more
critical than video for almost all applications, but its importance
(relative to other packets in the flow) is still an application decision.

Examples: Super bowl, On-Demand Streaming

Requirement:

Signal the flow needs least delay between server and client. Network can provide minimal delay information to the host. Feedback from the network and the client, based on user preference, is required for more efficient streaming. This is required for incorporating special needs based on application/user capabilities and to prioritize traffic during reactive events.

Problems:

1.  All packets prioritized the same irrespective of user preferences/needs:

>     a.  A client based change in priority of a certain type of data is not possible that. Eg. A hearing challenged user can choose video over audio while the priority is different for a regular user.

Reviewer: ```suggestion
    a.  A client based change in priority of a certain type of data is not possible. For example, a hearing challenged user can choose video over audio while the priority is different for other users.
```

Reviewer: to be more inclusive.

>     b.  Dynamic changes to priority based on user activity is not possible today. Eg. Audio packets having same priority when user mutes the audio, or change in priority during time of emergency where video streaming applications share the same priority as SOS signals.

Reviewer: ```suggestion
    b.  Dynamic changes to priority based on user activity is not possible today. For example, audio packets having the same priority when a user mutes the audio locally, or change in priority during time of emergency where video streaming applications share the same priority as SOS signals.
```

> 2.  In loss prone network or during a reactive event, retransmissions cause immense delay. Network, not able to distinguish between reliable and loss tolerant data or the critical data within flows(i-frames, p-frames and audio packets), can have challenges in efficiently handling/routing data.

Reviewer: ```suggestion
2.  In loss-prone networks or during a reactive policy events, retransmissions cause immense delay. Networks, not able to distinguish between reliable and loss-tolerant data or the critical data within a flow (i-frames, p-frames, and audio packets), can have challenges in efficiently handling/forwarding data.
```

Solution:

1.  Client to ISP (Host-to-network) signaling can help with conveying user needs.

>     a. The server (content provider) achieves best scalability by sending a single stream with the same per-packet metadata. The client, on the other hand, signaling the ISP to treat certain packets with a different priority over the others, can help solve the issue. Eg. For video streaming, if the metadata just said "audio" (0x00), "video i-frame" (0x01) "video p-frame" (0x02), the client could have the signaling protocol tell the router which one was most important. Some users could prioritize video over audio and vice versa.

Reviewer: ```suggestion
    a. The server (content provider) achieves best scalability by sending a single stream with the same per-packet metadata. The client, on the other hand, signaling the ISP to treat certain packets with a different priority over the others, can help solve the issue. For example, for video streaming, if the metadata just said "audio" (0x00), "video i-frame" (0x01) "video p-frame" (0x02), the client could have the signaling protocol tell an upstream router which one was most important. Some users could prioritize video over audio and vice versa.
```

resources cannot be used at maximum and, thus, a network would seek to
reduce or offload some of the traffic during these events -- often

> called 'reactive traffic policy'. An example of such sue case is cellular networks

Reviewer 2: replace "sue" with "such"

resources cannot be used at maximum and, thus, a network would seek to
reduce or offload some of the traffic during these events -- often

> called 'reactive traffic policy'. An example of such sue case is cellular networks

Reviewer: ```suggestion
called 'reactive traffic policy'. An example of such use case is cellular networks
```

that are overly used (and radio resources exhausted) such as a large
collection of people (e.g., parade, sporting event), or such as a partial

> radio network outage (e.g., tower power outage).  During such an condition, an alternative network

Reviewer: ```suggestion
radio network outage (e.g., tower power outage).  During such a condition, an alternative network
```
